src/plugins/knex.js

```diff
@@ -2,22 +2,145 @@
 
 const tx = require('./util/promise')
 
-function createPatch (file) {
-  return {
-    name: 'knex',
-    versions: ['>=0.8.0'],
-    file,
-    patch (Builder, tracer, config) {
-      this.wrap(Builder.prototype, 'then', tx.createWrapThen(tracer, config))
-    },
-    unpatch (Builder) {
-      this.unwrap(Builder.prototype, 'then')
+const spanSymbol = '_sfxSpan'
+
+function createWrapBuilder (tracer, config) {
+  return function wrapQueryBuilder (original) {
+    return function queryBuilderWithTrace () {
+      const scope = tracer.scope()
+      const span = scope.active()
+      const builder = original.apply(this, arguments)
+      builder[spanSymbol] = span
+      return builder
+    }
+  }
+}
+
+function createWrapRunner (wrapper, tracer, config) {
+  return function wrapRunner (original) {
+    return function runnerWithTrace () {
+      const runner = original.apply(this, arguments)
+      wrapper.wrap(runner, 'query', createWrapRunnerQuery(tracer, config))
+      return runner
+    }
+  }
+}
+
+function createWrapRunnerQuery (tracer, config) {
+  return function wrapQuery (original) {
+    return function queryWithTrace (q) {
+      const scope = tracer.scope()
+      const childOf = this.builder[spanSymbol]
+
+      const tags = {
+        'component': 'knex',
+        'db.statement': q.sql
```

Comment:
Mind truncating to 1024 chars to prevent size issues?

Reply:
I wouldn't. This doesn't include any "data" unless someone manually compiles queries and then uses them in an raw operation. Historically we haven't tried to enforce any sizing restrictions on tags at least, not on the client side. Happy to hear other thoughts.

Reply:
The SFx offerings all (should) truncate db.statements to 1024 chars for its current [size restrictions](https://docs.signalfx.com/en/latest/apm/apm-metadata/apm-metadata.html#limits-on-span-metadata).  When we don't users have run into difficulties that are inconvenient to track down.

Reply:
Done